File: app/src/main/java/it/cammino/risuscito/GeneralSearch.java
package it.cammino.risuscito;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.design.widget.TabLayout;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;
import android.support.v4.view.ViewPager;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import java.util.Locale;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.Unbinder;
import it.cammino.risuscito.ui.ThemeableActivity;

public class GeneralSearch extends Fragment {

  @BindView(R.id.view_pager)
  ViewPager mViewPager;

  private Unbinder mUnbinder;

  @Override
  public View onCreateView(
      @NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
    View rootView = inflater.inflate(R.layout.activity_general_search, container, false);
    mUnbinder = ButterKnife.bind(this, rootView);

    MainActivity mMainActivity = (MainActivity) getActivity();
    mMainActivity.setupToolbarTitle(R.string.title_activity_search);

    mViewPager.setAdapter(new SectionsPagerAdapter(getChildFragmentManager()));

    final TabLayout tabs = mMainActivity.mTabLayout;
    tabs.setVisibility(View.VISIBLE);
    mMainActivity.enableFab(false);
    if (!mMainActivity.isOnTablet()) mMainActivity.enableBottombar(false);
    tabs.setupWithViewPager(mViewPager);

    return rootView;
  }

  @Override
  public void onDestroyView() {
    super.onDestroyView();
    mUnbinder.unbind();
  }

  private class SectionsPagerAdapter extends FragmentPagerAdapter {

    SectionsPagerAdapter(FragmentManager fm) {
      super(fm);
    }

    @Override
    public Fragment getItem(int position) {
      switch (position) {
        case 0:
          return new RicercaVeloceFragment();
        case 1:
          return new RicercaAvanzataFragment();
        default:
          return new RicercaVeloceFragment();
      }
    }

    @Override
    public int getCount() {
      return 2;
    }

    @Override
    public CharSequence getPageTitle(int position) {
      Locale l =
          ThemeableActivity.getSystemLocalWrapper(getActivity().getResources().getConfiguration());
      switch (position) {
        case 0:
          return getString(R.string.fast_search_title).toUpperCase(l);
        case 1:
          return getString(R.string.advanced_search_title).toUpperCase(l);
        default:
          return "";
      }
    }
  }
}

File: app/src/main/java/it/cammino/risuscito/GeneralSearch.kt
package it.cammino.risuscito

import android.os.Bundle
import android.preference.PreferenceManager
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import androidx.lifecycle.ViewModelProviders
import it.cammino.risuscito.ui.ThemeableActivity
import it.cammino.risuscito.viewmodels.GeneralIndexViewModel
import kotlinx.android.synthetic.main.tabs_layout.*

class GeneralSearch : Fragment() {

    private var mMainActivity: MainActivity? = null

    private var mViewModel: GeneralIndexViewModel? = null

    override fun onCreateView(
            inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {

        mViewModel = ViewModelProviders.of(this).get(GeneralIndexViewModel::class.java)

        mMainActivity = activity as MainActivity?
        mMainActivity!!.setupToolbarTitle(R.string.title_activity_search)

        return inflater.inflate(R.layout.activity_general_search, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        view_pager.adapter = SectionsPagerAdapter(childFragmentManager)

        mMainActivity!!.setTabVisible(true)
        mMainActivity!!.enableFab(false)
        mMainActivity!!.enableBottombar(false)
        if (savedInstanceState == null) {
            val pref = PreferenceManager.getDefaultSharedPreferences(context)
            view_pager.currentItem = Integer.parseInt(pref.getString(Utility.DEFAULT_SEARCH, "0")!!)
        } else
            view_pager.currentItem = mViewModel!!.pageViewed
        mMainActivity!!.getMaterialTabs().setupWithViewPager(view_pager)
    }

    override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)
        mViewModel!!.pageViewed = view_pager.currentItem
    }

    private inner class SectionsPagerAdapter internal constructor(fm: FragmentManager) : FragmentPagerAdapter(fm) {

        override fun getItem(position: Int): Fragment {
            return when (position) {
                0 -> RicercaVeloceFragment()
                1 -> RicercaAvanzataFragment()
                else -> RicercaVeloceFragment()
            }
        }

        override fun getCount(): Int {
            return 2
        }

        override fun getPageTitle(position: Int): CharSequence? {
            val l = ThemeableActivity.getSystemLocalWrapper(activity!!.resources.configuration)
            return when (position) {
                0 -> getString(R.string.fast_search_title).toUpperCase(l)
                1 -> getString(R.string.advanced_search_title).toUpperCase(l)
                else -> ""
            }
        }
    }
}
